packages/next-urql/src/with-urql-client.tsx

```diff
@@ -50,54 +51,56 @@ export function withUrqlClient(
     // Set the displayName to indicate use of withUrqlClient.
     withUrql.displayName = `withUrqlClient(${getDisplayName(AppOrPage)})`;
 
-    withUrql.getInitialProps = async (appOrPageCtx: NextUrqlContext) => {
-      const { AppTree } = appOrPageCtx;
-
-      // Determine if we are wrapping an App component or a Page component.
-      const isApp = !!(appOrPageCtx as AppContext).Component;
-      const ctx = isApp
-        ? (appOrPageCtx as AppContext).ctx
-        : (appOrPageCtx as NextPageContext);
-
-      const opts =
-        typeof clientConfig === 'function' ? clientConfig(ctx) : clientConfig;
-      const [urqlClient, ssrCache] = initUrqlClient(opts, mergeExchanges);
-
-      if (urqlClient) {
-        (ctx as NextUrqlContext).urqlClient = urqlClient;
-      }
-
-      // Run the wrapped component's getInitialProps function.
-      let pageProps = {} as any;
-      if (AppOrPage.getInitialProps) {
-        pageProps = await AppOrPage.getInitialProps(appOrPageCtx as any);
-      }
-
-      // Check the window object to determine whether or not we are on the server.
-      // getInitialProps runs on the server for initial render, and on the client for navigation.
-      // We only want to run the prepass step on the server.
-      if (typeof window !== 'undefined') {
-        return { ...pageProps, urqlClient };
-      }
-
-      const props = { ...pageProps, urqlClient };
-      const appTreeProps = isApp ? props : { pageProps: props };
-
-      // Run the prepass step on AppTree. This will run all urql queries on the server.
-      await ssrPrepass(<AppTree {...appTreeProps} />);
-
-      // Serialize the urqlClient to null on the client-side.
-      // This ensures we don't share client and server instances of the urqlClient.
-      (urqlClient as any).toJSON = () => {
-        return null;
-      };
+    if (AppOrPage.getInitialProps || !noSsr) {
```

Comment:
if we don't see a `getInitialProps` or we haven't disabled this subtree from `getInitialProps` we bind it to `withUrql` else the consumer of our library can export `getServersideProps` or `getStaticProps`

Reply:
hey Jovi -- so as a consumer i would have to opt-in to automatic static optimization?

Reply:
Hey, so the problem is that before we had no check so I added an opt-in check for backwards compatability but the goal would be to just rely in them opting in to tree-traversal.

This opt-in step is meant for the user to declare they are using `getStaticProps` or `getServersideProps`

Reply:
oh i see so it's for those on versions of Next without the automatic optimization to opt in to it?

Reply:
yes

Reply:
So the thing is that if we add `getInitialProps` in the tree and the user exports `getStaticProps` or `getServerSideProps` next will throw an error saying that these shouldn't be combined. So we allow for the user to tell us to skip out on us adding this to `withUrql`

Reply:
oh that makes sense, thanks for clarifying :)